### DISCUSSION TITLE
Suggested way to implement for multi-module apps

### DISCUSSION THREAD

**Discussion by sudhirkhanger** [PatilShreyas/compose-report-to-html#61] [Q&A]
How can I implement this in a multi-module app? My current workflow is generating all the metrics in a single place and then running the CLI which gives me a nice single index.html. Is something like that possible with the Gradle plugin?

It seems Gradle plugin is `Apply in the module level project:` applied at module level and it generates reports and html individually for each module.

**Comment by PatilShreyas** [comment]
> How can I implement this in a multi-module app? My current workflow is generating all the metrics in a single place and then running the CLI which gives me a nice single index.html. Is something like that possible with the Gradle plugin?
> 
> It seems Gradle plugin is `Apply in the module level project:` applied at module level and it generates reports and html individually for each module.

@sudhirkhanger correct. The current Gradle plugin does not support the aggregation of modules. We can try to build this.

**Comment by satheshrgs-rksv** [comment]
Is this supported now ? I am also trying on a multi module project. Also i tried to use the CLI and it seems like it needs JAVA 11 ? 


Got this error for npm cli
Error: LinkageError occurred while loading main class dev.shreyaspatil.composeCompilerMetricsGenerator.cli.MainKt
        java.lang.UnsupportedClassVersionError: dev/shreyaspatil/composeCompilerMetricsGenerator/cli/MainKt has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0
